### PR TITLE
Allow the Agent to retrieve configuration from the Fleet API.

### PR DESCRIPTION
**Pull request by ph**
Correctly Fetch a configuration from the Fleet API and send everything to the operator.
This PR doesn't include errors reportings.

Ref: #14951

depends on: #15089

--- 

How to test?

- Use the dev-tools/cmd/fakewebapi
- sh push.sh to push a configuration to the API.
- mage build:dev
- ./agent enroll http://localhost:8080 abc123
- Adjust the path in agent.yml see #15167 for followup.
- BEATS_VERSION=7.5.0 agent run

### CODE REVIEW

**Comment by michalpristas** [comment]
@ph if you fix `// +build` directives as suggested build will pass

**Comment by ph** [comment]
@michalpristas  you can a review on this.

File: x-pack/agent/pkg/agent/application/fleet_gateway.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+		select {
+		case <-f.scheduler.WaitTick():
+			f.log.Debug("FleetGateway calling Checkin API")
+			resp, err := f.execute()
+			if err != nil {
+				f.log.Error(err)
+				continue
+			}
+
+			actions := make([]action, len(resp.Actions))
```
Comment (michalpristas) [inline x-pack/agent/pkg/agent/application/fleet_gateway.go]: why do you need to repack this?

File: x-pack/agent/pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	C chan time.Time
+}
+
+// Next trigger the WaitTick unblock manually.
+func (s *Stepper) Next() {
+	s.C <- time.Now()
+}
+
+// WaitTick returns a channel to watch for ticks.
+func (s *Stepper) WaitTick() <-chan time.Time {
```
[the remark this answers is not in the document]
Comment (ph) [reply x-pack/agent/pkg/scheduler/scheduler.go]: @michalpristas what if we use Tick() instead?

File: x-pack/agent/pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+// WaitTick wait on the duration to be experied to unblock the channel.
+// Note: you should not keep a reference to the channel.
+func (p *Periodic) WaitTick() <-chan time.Time {
+	if p.ran {
+		return p.Ticker.C
+	}
+
+	rC := make(chan time.Time)
```
[the remark this answers is not in the document]
Comment (ph) [reply x-pack/agent/pkg/scheduler/scheduler.go]: good point I will do that.

File: x-pack/agent/pkg/agent/application/managed_mode.go [hunk 4]
```diff
@@ -9,18 +9,21 @@ import (
 	"net/http"
 	"net/url"
 
+	"time"
```
[the remark this answers is not in the document]
Comment (ph) [reply x-pack/agent/pkg/agent/application/managed_mode.go]: looks fine, I've ran `make fmt` and nothing was raised.

File: x-pack/agent/pkg/agent/application/fleet_gateway.go (hunk 1 above)
Comment (ph) [reply x-pack/agent/pkg/agent/application/fleet_gateway.go]: Because the `actionDispatcher` doesn't have hard dependencies on `fleetapi.Action`

**Comment by ph** [comment]
@michalpristas updated with all the changes.

**Comment by ph** [comment]
@michalpristas I have addressed all the above,  merging this to unblock agent uuid.

**Pull request merged**